### PR TITLE
Stages

### PR DESCRIPTION
**Pull request by aronnoordam**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -29,10 +29,10 @@ def set_install_requirements():
         python_version_part = '-cp311-cp311'
 
     requirements = [
-        f"KratosMultiphysics @ https://github.com/StemVibrations/StemKratos/raw/main/StemKratos/wheels/KratosMultiphysics-{kratos_version}{python_version_part}{platform_part}",
-        f"KratosLinearSolversApplication @ https://github.com/StemVibrations/StemKratos/raw/main/StemKratos/wheels/KratosLinearSolversApplication-{kratos_version}{python_version_part}{platform_part}",
-        f"KratosStructuralMechanicsApplication @ https://github.com/StemVibrations/StemKratos/raw/main/StemKratos/wheels/KratosStructuralMechanicsApplication-{kratos_version}{python_version_part}{platform_part}",
-        f"KratosGeoMechanicsApplication @ https://github.com/StemVibrations/StemKratos/raw/main/StemKratos/wheels/KratosGeoMechanicsApplication-{kratos_version}{python_version_part}{platform_part}"
+        f"KratosMultiphysics @ https://github.com/StemVibrations/StemKratos/raw/stages/StemKratos/wheels/KratosMultiphysics-{kratos_version}{python_version_part}{platform_part}",
```
Comment (aronnoordam) [inline setup.py:32]: make sure to revert this branch ref before merge

File: tests/test_call_uvec_multistage.py [hunk 2]
```diff
@@ -17,37 +17,19 @@ def test_call_uvec_multi_stage():
...
     calculated_disp_file = Path(r"tests/test_data/input_data_multi_stage_uvec/output/calculated_disp")
     expected_disp_file = Path(r"tests/test_data/input_data_multi_stage_uvec/_output/expected_disp")
 
     # check if calculated disp below first wheel is equal to expected disp
-    are_files_equal = assert_floats_in_files_almost_equal(calculated_disp_file, expected_disp_file)
+    are_files_equal, message = assert_floats_in_files_almost_equal(calculated_disp_file, expected_disp_file)
+
     # remove calculated disp file as data is appended
     calculated_disp_file.unlink()
-    assert are_files_equal
```
Comment (brunozc) [inline tests/test_call_uvec_multistage.py:50]: line 42 use os.path.join or Path insteal of /

btw why are you moving the files around?

File: tests/utils.py [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 from pathlib import Path
 
 import KratosMultiphysics as Kratos
 import KratosMultiphysics.StemApplication.geomechanics_analysis as analysis
 
 
-class Utils():
+class Utils:
     @staticmethod
     def run_stage(project_path, project_parameters_file_name="ProjectParameters.json"):
```
Comment (brunozc) [inline tests/utils.py:11]: missing types

File: tests/utils.py [hunk 4]
```diff
@@ -35,14 +35,46 @@ def run_stage(project_path, project_parameters_file_name="ProjectParameters.json
...
+            # read parameters
+            with open(file_name, 'r') as parameter_file:
+                parameters = Kratos.Parameters(parameter_file.read())
+
+            # run stage
+            stage = analysis.StemGeoMechanicsAnalysis(model, parameters)
+            stage.Run()
+
+            # change working directory back to original working directory
+            os.chdir(cwd)
```
Comment (brunozc) [inline tests/utils.py:68]: why is run_stage returning and run_multiple_stages not returning? 
Also both functions could be merged. You iterate around project_parameters_files. In the case of one stage this is one loop

**Review by brunozc — APPROVED** [review]

**Review by brunozc — APPROVED** [review]

**Pull request merged**